ui/app/AppMain.qml

```diff
@@ -531,78 +533,80 @@ Item {
                                 // we cannot return QVariant if we pass another parameter in a function call
                                 // that's why we're using it this way
                                 mainModule.prepareCommunitySectionModuleForCommunityId(model.id)
-                                store = mainModule.getCommunitySectionModule()
+                                chatCommunitySectionModule = mainModule.getCommunitySectionModule()
```

Comment:
@alexandraB99 will you tell me what do you have on your mind, what's your idea (to have loading on demand or)?

Reply:
@saledjenic using the loader I introduced [here](https://github.com/status-im/status-desktop/blob/e666d4d6bf69a1cf53855c78f999567562a5a77a/ui/app/AppMain.qml#L432), did you test that? did it work?

Reply:
> did you test that?

@alexandraB99 no I didn't, I looked at the code and saw that solution wouldn't fit the case when we store current index in order to open the app displaying the same section which was active before closing the app. Also in that case stack layout children indexes may not follow the ordering we have in left app nav bar and that's somehow important for other features we have since then the index in the stack layout will depend on the order in which you're switching sections. Also maybe there are some features in components beneath the top components added to that stack layout which may depend on indexes. All in all for having a communities loading on demand I think we would need to do more updates on the qml side, in that case all children of stack layout may be loaded using loader, but we may do that in some of our next refactoring phase.